doc/source/WorkingWithThePythonScriptingAPI/ThePythonEditor/index.md

```diff
@@ -0,0 +1,78 @@
+```eval_rst
+.. role:: raw-html(raw)
+    :format: html
+```
+
+# The Python Editor #
+
+The _Python Editor_ is a type of editor in the main Gaffer application that can execute arbitrary Python code with a built-in interpreter. It enables you to modify the node graph, try commands and syntax, and inspect data. In the default layout, it resides in the bottom-right panel, in a tab next to the _Hierarchy View_.
+The _Python Editor_ is split into two areas. The bottom-half is the input field. The top-half is the output. The input field of the _Python Editor_ functions like a basic plain text editor. You can type, select, cut, copy, paste, and execute code.
+
+![A Python Editor with its input and the output field](images/interfacePythonEditorOutput.png "A Python Editor with its input and the output field")
+
+The output field behaves like a typical console output. It records the output stream, including return values and errors. Errors appear in red.
+
+![Errors in a Python Editor's output field](images/interfacePythonEditorError.png "Errors in a Python Editor's output field")
+
+> Tip :
+> When adding a node to the graph using the _Python Editor_, you must first `import` the node's module, which will persist for the lifetime of the editor. However, you never need to `import` the [Gaffer](../../Reference/NodeReference/Gaffer/index.md) or [IECore](../../Reference/NodeReference/IECore/index.md) modules. They are loaded by default.
+
+> Caution :
+> Code in Expression nodes is handled by a separate interpreter, and the syntax between them and the _Python Editor_ should not be considered safely compatible.
+
+
+## Instructions ##
+
+
+### Executing code ###
+
+Executing code will erase the input field, just like in a debug console. However, if you highlight any part of the code, only that part will execute, and the whole input will be preserved.
+
+To execute code with the _Python Editor:_
+
+1. Enter code into the input field.
+2. Hit <kbd>Control</kbd> + <kbd>Enter</kbd>.
+
+> Tip :
+> A useful debugging trick is to highlight any part of the input code in order to preserve the whole input for reuse.
+
+
```

Comment:
It might be nice to add some links to the Scripting Tutorials/Reference here?

Reply:
The plan is that as soon as this is merged, I'll make a separate PR that moves all the scripting tutorials into this section, and then add links. Doing it this way just saves the headache of doubling up.